pkg/controller/change/network/watcher.go

```diff
@@ -119,25 +139,40 @@ func (w *DeviceWatcher) Start(ch chan<- types.ID) error {
 	return nil
 }
 
-// watchDevice watches changes for the given device
-func (w *DeviceWatcher) watchDevice(deviceID device.VersionedID, ch chan<- types.ID) {
+// updateWatch watches changes for the given device
+func (w *DeviceWatcher) updateWatch(topodevice *devicetopo.Device, ch chan<- types.ID) {
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
+	deviceID := device.NewVersionedID(device.ID(topodevice.ID), device.Version(topodevice.Version))
+	log.Infof("Updating watch for device %v", deviceID)
+
+	// If the protocol state is connected, ensure a stream is open
+	// If the protocol state is not connected, close any stream that's open
 	ctx := w.streams[deviceID]
-	if ctx != nil {
-		log.Errorf("Network DeviceWatcher for device changes %s - ctx already found", deviceID)
+	state := getProtocolState(topodevice)
```

Comment:
this does not allow for the watcher to start for a device which is not present in TOPO